python_hw_1.py

```diff
@@ -0,0 +1,97 @@
+#! /usr/bin/python 
+
+
+def transcribe(seq):
+    """Prints the transcribed sequence"""
+    transcription_seq = ""
+    transcription_dic = {"A" : "U", "T" : "A", "G" : "C", "C" : "G", 
+                "a" : "u", "t" : "a", "g" : "c", "c" : "g"}
+    for letter in seq:
+        if letter not in transcription_dic.keys():
+            print("Invalid alphabet")
+            break
+        else:
+            transcription_seq += transcription_dic[letter]
+    print(transcription_seq)
+    return transcription_seq 
+
+def reverse(seq):
+    """Prints the reverse sequence"""
+    rev_seq = ""
+    for letter in seq:
+        rev_seq = seq[::-1]
+    print(rev_seq)
+    return rev_seq 
```

Comment:
Здесь ты для каждого нуклеотида выполняешь одну и ту же операцию реверса для **всей** последовательности. По сути просто повторяется много раз ненужная работа, достаточно одного

```suggestion
    return seq[::-1] 
```